tasmota/xsns_88_tfminiplus.ino

```diff
@@ -0,0 +1,246 @@
+    {
+        crc += (uint16_t)Tfmp_buffer[i];
+    }
+    if (!(char)(crc & 0xff) != Tfmp_buffer[TFMP_MAX_DATA_LEN])
+    {
+        DEBUG_SENSOR_LOG(PSTR("TFmini Plus: crc error"));
+    }
+    DEBUG_SENSOR_LOG(PSTR("TFmini Plus: Distance: %d"), tfminiplus_sensor.distance);
+    DEBUG_SENSOR_LOG(PSTR("TFmini Plus: Signal: %d"), tfminiplus_sensor.sigstrength);
+    DEBUG_SENSOR_LOG(PSTR("TFmini Plus: Chip Temp: %d"), tfminiplus_sensor.chiptemp);
+    TfmpSerial->flush();
+}
+
+void TfmpProcessSerialData (void)
+{
+    uint8_t data;
+    bool dataReady;
+    if (TfmpSerial && tfminiplus_sensor.ready)
+    {
+        while (TfmpSerial->available() > 0) 
+        {
+            data = TfmpSerial->read();
+            dataReady = TfmpAddData((char)data);
+            if (dataReady)
+            {
+                TfmpProcessData();
+            }
+        }
+    }
+}
+
+bool TfmpAddData(char nextChar)
+{  
+    // Buffer position
+    static uint8_t currentIndex = 0;
+    // Store data into buffer at position
+    Tfmp_buffer[currentIndex] = nextChar;
+    currentIndex++;
+    // Check for too many data
+    if (currentIndex >= TFMP_MAX_DATA_LEN)
+    {
+        // Terminate buffer and reset position
+        Tfmp_buffer[TFMP_MAX_DATA_LEN] = '\0';
+        currentIndex = 0;
+        return true;
+    }
+    return false;
+}
+
+#ifdef USE_WEBSERVER
+// {s} = <tr><th>, {m} = </th><td>, {e} = </td></tr>
+const char HTTP_SNS_DISTANCE_CM[] PROGMEM = "{s}   " D_DISTANCE "{m}%d" D_UNIT_CENTIMETER "{e}";
+const char HTTP_SNS_SIGNALSTRENGTH[] PROGMEM = "{s}   " D_SIGNALSTRENGTH "{m}%d{e}";
+const char HTTP_SNS_CHIPTEMPERATURE[] PROGMEM = "{s}   " D_CHIPTEMPERATURE "{m}%d" D_UNIT_DEGREE D_UNIT_CELSIUS "{e}";
+#endif  // USE_WEBSERVER
+
+void TfmpShow(bool json)
+{
+    if (json) {
+        ResponseAppend_P(PSTR(",\"TFmini Plus\":{\"" D_JSON_DISTANCE "\":\"%d\",\"" D_JSON_SIGNALSTRENGTH "\":\"%d\",\"" D_JSON_CHIPTEMPERATURE "\":%d}"),
+            tfminiplus_sensor.distance, tfminiplus_sensor.sigstrength, tfminiplus_sensor.chiptemp);
+#ifdef USE_DOMOTICZ
+        if (0 == TasmotaGlobal.tele_period) {
+            DomoticzFloatSensor(DZ_COUNT, tfminiplus_sensor.distance);
+        }
+#endif  // USE_DOMOTICZ
+#ifdef USE_WEBSERVER
+    } else {
+        WSContentSend_P("{s}TFmini Plus");
+        WSContentSend_P(HTTP_SNS_DISTANCE_CM, tfminiplus_sensor.distance);
+        WSContentSend_P(HTTP_SNS_SIGNALSTRENGTH, tfminiplus_sensor.sigstrength);
+        WSContentSend_P(HTTP_SNS_CHIPTEMPERATURE, tfminiplus_sensor.chiptemp);
+#endif  // USE_WEBSERVER
+    }
+}
+
+/*********************************************************************************************\
+ * Interface
+\*********************************************************************************************/
+
+bool Xsns88(byte callback_id)
+{
+    bool result = false;
+    switch (callback_id) {
+        case FUNC_INIT:
+            TfmpInit();
+            break;
+        case FUNC_EVERY_SECOND:
```

Comment:
I'll suggest to the presence check 
```
if (TfmpSerial && tfminiplus_sensor.ready)
```
here. This way all but FUNC_INIT are ignored when no tfmini is connected. See ``xsns_85_mpu6886.ino`` how to.

Reply:
Proposal included according example.

Reply:
The switch function isn't anymore so nice as from your guide line proposal...